src/test/java/org/jenkinsci/plugins/githubautostatus/ConfigurationAsCodeTest.java

```diff
@@ -0,0 +1,57 @@
+package org.jenkinsci.plugins.githubautostatus;
+
+import hudson.ExtensionList;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.ConfiguratorRegistry;
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import io.jenkins.plugins.casc.model.CNode;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static io.jenkins.plugins.casc.misc.Util.*;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.*;
+
+public class ConfigurationAsCodeTest {
+
+    @ClassRule
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public static JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    public void should_support_configuration_as_code() {
+        BuildStatusConfig config = ExtensionList.lookupSingleton(BuildStatusConfig.class);
+        assertNotNull(config);
+        assertEquals(2, config.getDbVersion().intValue());
```

Comment:
I suggest to sort the assertions here in the same order as they occur in the YAML file. This should make it easier to find/compare them.